### DISCUSSION TITLE
Forking and customizing questions

### DISCUSSION THREAD

**Discussion by live4thamuzik** [prasanthrangan/hyprdots#2030] [Q&A]
Hello,

First and foremost, thank you for these dots and the script to install HyDe! Great work, very inspiring! 

I am quite new to using linux and hyprland as a daily driver (been about 6-9 months) coming from 20+ years of being a windows user. I dabbled in Mac OS but that was very short lived (maybe 4yrs before I couldn't stand the walled garden that is Apple anymore). 

I want to fork this project and hopefully, (fingers crossed) figure out a way to add the fork to my first shell script I put together for installing Arch Linux with minimal configuration. There a couple things I'd elect to change in the fork like; keeping bash and using oh-my-posh rather than zsh, fish, lsd, starship, oh-my-zsh, etc...

I also change a few things like; keybinds, having SDDM, Waybar, and hyprlock using 24hr clock vs 12hr, swapping out wallpapers, etc. One problem I have experienced is, even though I comment out lines in the script(s) for systemd-boot, it will still try to decrypt my lvm using systemd, fail because it cant find systemd-boot or the drive "hd0" then will proceed to the correct screen for decrypting LUKS as defined in /etc/default/grub. This could all be avoided of course if I opt of of the grub theme during install then add it manually later like I have done in the past. Additionally I'd like to add in plymouth for splash screens but that I can sort out on my own.

Before I do any of this, I'd like a little clarification on forking repos and changing the code to my liking as I am also new to having repos on github and have not quite figured all of this out. I know some of these questions may seem stupid or elementary but I'd rather ask and get roasted vs. not ask and assume which never ends well... 

IF I fork this and start changing things and committing changes to the fork, does that write back to this project? If not when this repo is updated and I want to keep the fork current, am I overwriting all customizations I made? I have read all scripts in great detail and understand most but get a little confused with the amount of variables being called. Is there a better way of omitting the shell change and the sysytemd-boot stuff I don't want? I'd be very grateful for any advise on removing those particular features. Is it safer/better to just clone the repo and try to make my own version with the dots I use on my daily driver? 

Basically I am looking for a way to be able replicate the setup I have on any machine by simply using my shell script and maybe a fork of this repo. 

Thanks for your time, Cheers!

**Comment by kRHYME7** [comment]
Push it to your fork.
Fork this in github.
clone your fork locally 
Do changes
git add .
git commit 
git push remotely 

-  i only read the last paragraph.

**Reply by live4thamuzik** [reply]
Thanks for responding so quickly! I totally get the TLDR and jumping to last paragraph. I'll try the fork and follow your guidance on how to properly manage the fork.
